### PR TITLE
regenmeter: notif before next hp regen & refactor existing code to use Constants.GAME_TICK_LENGTH

### LINKED ISSUE #2704: Add Health Regeneration Notification
NMZ suggestion: Making a notification appear whenever your health is about regenerate at a certain point such as at this point in the regeneration process. 
hp sugg

### PR DESCRIPTION
**Pull request by seandewar**
Adds a configuration option for notifying the player before their next HP regen occurs, and refactors existing code to utilise `Constants.GAME_TICK_LENGTH` where approperiate.

The following notification is sent:

Should be especially useful for NMZ. Its accuracy is limited to that of the Regeneration Meter plugin itself.
Closes #2704.

**Note:** the config accepts the amount of milliseconds before the regen to send the notification, but this is instead converted to the earliest corresponding 0.6s tick interval in the code. Maybe it would be a better idea to accept something like seconds instead?

### CODE REVIEW

**Comment by seandewar** [comment]
After a small discussion on the discord, I think it's better for the config option to accept seconds instead of milliseconds. I've modified the PR to reflect that change.

File: runelite-client/src/main/java/net/runelite/client/plugins/regenmeter/RegenMeterConfig.java [hunk 1]
```diff
@@ -57,4 +57,13 @@ default boolean showWhenNoChange()
 	{
 		return false;
 	}
+
+	@ConfigItem(
+		keyName = "notifyBeforeHpRegenDuration",
+		name = "Hitpoint Regen Notification (seconds)",
+		description = "Notify approximately when your next hitpoint is about to regen. A value of 0 will disable notification.")
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/regenmeter/RegenMeterConfig.java]:
```suggestion
		description = "Notify approximately when your next hitpoint is about to regen. A value of 0 will disable notification."
	)
```

File: runelite-client/src/main/java/net/runelite/client/plugins/regenmeter/RegenMeterPlugin.java [hunk 2]
```diff
@@ -149,5 +155,18 @@ else if (currentHP > maxHP)
...
+		{
+			notifier.notify("[" + client.getLocalPlayer().getName() + "] regenerates their next hitpoint soon!");
+		}
+	}
+
+	private boolean shouldNotifyHpRegenThisTick(int ticksPerHPRegen)
+	{
+		// if the configured duration lies between two ticks, choose the earlier tick
+		final int ticksBeforeHPRegen = ticksPerHPRegen - ticksSinceHPRegen;
+		final int notifyTick = (int) Math.ceil(config.getNotifyBeforeHpRegenSeconds() / HP_REGEN_TICK_LENGTH);
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/regenmeter/RegenMeterPlugin.java]: This is basically game tick, don't we have constant for this somewhere? If no, then this is good opportunity to add it to Constants file

Comment (seandewar) [reply runelite-client/src/main/java/net/runelite/client/plugins/regenmeter/RegenMeterPlugin.java]: Yeah, I noticed this. I wasn't sure whether to add it seeing that a few plugins already define their own version of the 0.6s tick constant in a few places.

If I do this, should I also update the other plugins to use the const from Constants within this PR? I could add that change as a seperate commit to this one and edit the name of the PR to reflect this.

Comment (Nightfirecat) [reply runelite-client/src/main/java/net/runelite/client/plugins/regenmeter/RegenMeterPlugin.java]: A separate commit would be good, but that sounds like a good idea.

**Comment by seandewar** [comment]
Latest revision adds `GAME_TICK_LENGTH` to Constants.java and refactors existing code to refer to it instead of redefining the same value in multiple places.

I've also used `GAME_TICK_LENGTH` in places that I think seem to somewhat loosely refer to the 600ms game tick interval (including Hooks.java, whose main loop does some work every 600 ms -- I could use some feedback on whether that change is approperiate).

Other places, such as BlastMineRock.java's `FUSE_TIME` constant (=4.2s; exactly 7 game ticks), I've decided to leave alone, as these seem fine enough.

**Comment by seandewar** [comment]
@Adam- You want me to _flip_ the commits around?

**Comment by Adam-** [comment]
Yes

File: runelite-client/src/main/java/net/runelite/client/plugins/regenmeter/RegenMeterPlugin.java [hunk 3]
```diff
@@ -30,6 +30,7 @@
 import javax.inject.Inject;
 import lombok.Getter;
 import net.runelite.api.Client;
+import net.runelite.api.Constants;
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/regenmeter/RegenMeterPlugin.java]: ?

Comment (seandewar) [reply runelite-client/src/main/java/net/runelite/client/plugins/regenmeter/RegenMeterPlugin.java]: 🤦 whoops

**Comment by Adam-** [comment]
Better commit messages might be:

```
constants: add GAME_TICK_LENGTH and use where needed

Also update many usages of 20ms to CLIENT_TICK_LENGTH 
```

because 1) it is implied the commit modifies and codebase and 2) GAME_TICK_LENGTH is actually being added by the commit, not simply being used

and

`regenmeter: add option to notify before next hp regen`

**Pull request merged**